### PR TITLE
Plugins: Add sync update all plugins in cloud

### PR DESCRIPTION
**Pull request by oshirohugo**
**What is this feature?**

This PR make the update all requests synchronous in cloud. This is necessary because hosted grafana can't deal with multiple requests happening at the same time. (Check note for reviews for details)

**Why do we need this feature?**

This PR fixes the update all plugins feature in cloud.

**Who is this feature for?**

Grafana cloud users.

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/admin/components/UpdateAllModal.tsx [hunk 1]
```diff
@@ -215,24 +222,46 @@ export const UpdateAllModal = ({ isOpen, onDismiss, plugins }: Props) => {
...
       });
+
+      setInRequest('');
     }
   }, [error, errorMap, inProgress, selectedPlugins]);
 
+  // update plugins one by one (only used in cloud)
+  useEffect(() => {
+    if (startSyncUpdate && !inRequest) {
+      for (let plugin of plugins) {
```
Comment (academo) [inline public/app/features/plugins/admin/components/UpdateAllModal.tsx:233]: wouldn't this loop send all plugins to install in a single go anyway?

Comment (oshirohugo) [reply public/app/features/plugins/admin/components/UpdateAllModal.tsx:233]: we break the loop as soon as we get a selected plugin to install.

**Review by academo — APPROVED** [review]
much nicer!

**Pull request merged**